openverse_catalog/dags/common/storage/image.py

```diff
@@ -194,10 +197,19 @@ def __init__(
         logger.info(f"Initialized with provider {provider}")
         super().__init__(provider=provider)
         self.license_info = license_info
+        self.media_buffer = []
 
     def add_item(self, **kwargs):
         image_data = kwargs | {"thumbnail_url": None}
         for field in MockImageStore.NULLABLE_FIELDS:
             if field not in image_data:
                 image_data[field] = None
-        return self._get_image(**image_data)
+        image = self._get_image(**image_data)
+        if image is not None:
+            self.media_buffer.append(image)
+        return len(self.media_buffer)
+
+    def get_item(self):
+        if len(self.media_buffer):
+            return self.media_buffer[0]
+        return None
```

Comment:
I wonder if it might be best to just use `media_buffer` directly in tests, rather than a proxy method. That way if multiple items are added, the test can index into each one (or whichever one it needs). If an `IndexError` is raised, that'll fail the test too 😄

Reply:
Thank you, this will be much better!
This actually should be out of scope for this PR. Maybe I should actually create a new one.
My main goal was to test the ImageStore class the way it works, and not its inner implementation. Currently, many tests rely on the internal `_get_image` method to test that the image data is handled correctly. However, it is much more robust to test the available API methods, and not the internal implementation (which can and should be able to change).
We could probably add some helper methods to the MockImageStore class that would extract the image details from the `_media_buffer` list of TSV rows. I wonder if it would be better to make the `_media_buffer` in the `ImageStore` a public field...

Reply:
@AetherUnbound , I've decided that it's better to improve the tests in a different PR, and used `_clean_metadata` for the specific test.

Reply:
Sounds great!